doc/ring/ring.md

```diff
@@ -36,51 +73,73 @@ Applying the handler:
 ; {:status 200, :body "ok"}
 ```
 
-The expanded routes shows the compilation results:
+The router can be accessed via `get-router`:
 
 ```clj
-(-> app (ring/get-router) (reitit/routes))
-; [["/ping"
-;   {:handler #object[...]}
-;   #Methods{:any #Endpoint{:data {:handler #object[...]},
-;                           :handler #object[...],
-;                           :middleware []}}]]
+(-> app (ring/get-router) (r/compiled-routes))
+;[["/ping"
+;  {:handler #object[...]}
+;  #Methods{:get #Endpoint{:data {:handler #object[...]}
+;                          :handler #object[...]
+;                          :middleware []}
+;           :options #Endpoint{:data {:handler #object[...]}
+;                              :handler #object[...]
+;                              :middleware []}}]]
 ```
 
-Note the compiled resuts as third element in the route vector.
-
 # Request-method based routing
 
-Handler are also looked under request-method keys: `:get`, `:head`, `:patch`, `:delete`, `:options`, `:post` or `:put`. Top-level handler is used if request-method based handler is not found.
+Handlers can be placed either to the top-level (all methods) or under a spesific method (`:get`, `:head`, `:patch`, `:delete`, `:options`, `:post`, `:put` or `:trace`). Top-level handler is used if request-method based handler is not found. 
+
+By default, the `:options` route is generated for all paths - to enable thing like [CORS](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing).
 
 ```clj
 (def app
   (ring/ring-handler
     (ring/router
-      ["/ping" {:name ::ping
-                :get handler
-                :post handler}])))
+      [["/all" handler]
+       ["/ping" {:name ::ping
+                 :get handler
+                 :post handler}]])))
+```
+
+Top-level handler catches all methods:
+
+```clj
+(app {:request-method :delete, :uri "/all"})
+; {:status 200, :body "ok"}
+```
 
+Method-level handler catches only the method:
+
+```clj
 (app {:request-method :get, :uri "/ping"})
 ; {:status 200, :body "ok"}
 
 (app {:request-method :put, :uri "/ping"})
 ; nil
 ```
 
+By default, `:options` is also supported (ree router options to change this):
+
+```clj
+(app {:request-method :options, :uri "/ping"})
+; {:status 200, :body ""}
+```
+
 Name-based reverse routing:
 
 ```clj
 (-> app
     (ring/get-router)
-    (reitit/match-by-name ::ping)
-    :path)
+    (r/match-by-name ::ping)
+    (r/match->path))
 ; "/ping"
 ```
 
 # Middleware
 
-Middleware can be added with a `:middleware` key, either to top-level or under `:request-method` submap. It's value should be a vector of any the following:
+Middleware can be mounted using a `:middleware` key - either to top-level or under `:request-method` submap. It's value should be a vector of `reitit.middleware/IntoMiddleware` values. These include:
```

Comment:
`It's` -> `Its`. 

With "`:request-method` submap", do you mean under e.g. `:get` map? I think it'd be better to say "request method submap", then, because at first I thought `:request-method` is the concrete key for the map.